server/src/main/java/org/elasticsearch/search/aggregations/metrics/PercentilesAggregationBuilder.java

```diff
@@ -292,35 +292,39 @@ protected XContentBuilder doXContentBody(XContentBuilder builder, Params params)
     }
 
     @Override
-    protected boolean innerEquals(Object obj) {
+    public boolean equals(Object obj) {
+        if (this == obj) return true;
+        if (obj == null || getClass() != obj.getClass()) return false;
+        if (super.equals(obj) == false) return false;
+
         PercentilesAggregationBuilder other = (PercentilesAggregationBuilder) obj;
         if (!Objects.equals(method, other.method)) {
```

Comment:
While we're making changes, would you mind fixing this `!Objects.equals(...)` too?  Thanks!